systemtest/src/test/java/io/strimzi/systemtest/oauth/OauthAuthorizationST.java

```diff
@@ -0,0 +1,247 @@
+
+    }
+
+    @Description("As a team B, i has full access on topics starting with 'b-'")
+    @Test
+    void testTeamBWriteToTopic() throws IOException, KeyStoreException, ExecutionException, InterruptedException, TimeoutException {
+        LOGGER.info("Sending {} messages to broker with topic name {}", MESSAGE_COUNT, TOPIC_NAME);
+
+        LOGGER.info("Producer will not produce messages because authorization topic will failed. Team A can write only to topic starting with 'x-'");
+
+        try {
+            Future<Integer> invalidProducer = teamBOauthKafkaClient.sendMessagesTls(TOPIC_NAME, NAMESPACE, CLUSTER_NAME, TEAM_B_CLIENT,
+                    MESSAGE_COUNT, "SSL", TIMEOUT_SEND_RECV_MESSAGES);
+            invalidProducer.get(TIMEOUT_SEND_RECV_MESSAGES, TimeUnit.SECONDS);
+
+        } catch (Exception e) {
+            LOGGER.info("Excepted message:{}", e.getMessage());
+        }
+
+        LOGGER.info("Sending {} messages to broker with topic name {}", MESSAGE_COUNT, TOPIC_B);
+
+        Future<Integer> producer = teamBOauthKafkaClient.sendMessagesTls(TOPIC_B, NAMESPACE, CLUSTER_NAME, TEAM_B_CLIENT,
+                MESSAGE_COUNT, "SSL");
+
+        assertThat(producer.get(2, TimeUnit.MINUTES), is(MESSAGE_COUNT));
+
+        Future<Integer> consumer = teamBOauthKafkaClient.receiveMessagesTls(TOPIC_B, NAMESPACE, CLUSTER_NAME, TEAM_B_CLIENT,
+                MESSAGE_COUNT, "SSL", "x_" + CONSUMER_GROUP_NAME + "-" + rng.nextInt(Integer.MAX_VALUE));
+
+        assertThat(consumer.get(2, TimeUnit.MINUTES), is(MESSAGE_COUNT));
+    }
+
+    @Description("As a team A,  can write to topics starting with 'x-' and Team B can read from topics starting with 'x-'")
+    @Test
+    void testTeamAWriteToTopicStartingWithXAndTeamBReadFromTopicStartingWithX() throws IOException, KeyStoreException, InterruptedException, ExecutionException, TimeoutException {
+        // only write means that Team A can not create new topic 'x-.*'
+        String topicName = TOPIC_X + "-example";
+
+        KafkaTopicResource.topic(CLUSTER_NAME, topicName).done();
+
+        Future<Integer> producer = teamAOauthKafkaClient.sendMessagesTls(topicName, NAMESPACE, CLUSTER_NAME, TEAM_A_CLIENT,
+                MESSAGE_COUNT, "SSL");
+
+        assertThat(producer.get(1, TimeUnit.MINUTES), is(MESSAGE_COUNT));
+
+        Future<Integer> consumer = teamBOauthKafkaClient.receiveMessagesTls(topicName, NAMESPACE, CLUSTER_NAME, TEAM_B_CLIENT,
+                MESSAGE_COUNT, "SSL", "x_consumer_group_b");
+
+        assertThat(consumer.get(1, TimeUnit.MINUTES), is(MESSAGE_COUNT));
+    }
+
+    @Test
+    void testListTopics() {
+        // TODO: in the new PR add AdminClient support with operations listTopics(), etc.
+    }
+
+    @Test
+    void testClusterVerification() {
+        // TODO: create more examples via cluster wide stuff
+    }
+
+    @BeforeAll
+    void setUp()  {
+        LOGGER.info("Replacing all URIs");
```

Comment:
`validIssuerUri`, `jwksEndpointUri`,  `oauthTokenEndpointUri`. This one URIs, I know that it would be better to have an abstraction but this way you just change pointing Keycloak to another realm

Reply:
but of course, I will change this log message )